src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -303,8 +303,8 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
             widgetType: this.$$type,
           });
           isFirstPage =
-            helper.state.page === undefined ||
```

Comment:
I wonder if in the index widget, could we make the getter of helper.state warn? Or is that still used internally?

Reply:
There are still valid use cases for using `helper.state` in callbacks like `createURL()` or `refine()` to manipulate the latest helper state:

https://github.com/algolia/instantsearch.js/blob/7747c50c037b1a73cf74e9e82986a0c303822e5a/src/connectors/breadcrumb/connectBreadcrumb.ts#L204-L206

An ESLint rule should be able to catch that maybe? Not really trivial though. Ultimately we want the values from the main paths that are returned from `getWidgetRenderState` to be computed with `state`, not with `helper.state`, the rest is fine.